libs/header/p2p/exchange.go

```diff
@@ -224,9 +225,21 @@ func (ex *Exchange[H]) performRequest(
 		return nil, fmt.Errorf("no trusted peers")
 	}
 
-	//nolint:gosec // G404: Use of weak random number generator
-	index := rand.Intn(len(ex.trustedPeers))
-	return ex.request(ctx, ex.trustedPeers[index], req)
+	for ctx.Err() == nil {
+		//nolint:gosec // G404: Use of weak random number generator
+		index := rand.Intn(len(ex.trustedPeers))
+		cctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
why context.Background here?

Reply:
Does not have to be but it better be

Reply:
Why not just use `ex.ctx` here like we do everywhere else throughout?

Reply:
I mean that it's better to be a passed context, not a background. Changing it to `ctx` though, not `ex.ctx`